extract_data.py

```diff
@@ -49,6 +49,17 @@ def extract_and_collect_data_from_generated_files(directory: str, has_dedicated_
         for wifi_card in mobo[1:]:
             wifi_cards.append(wifi_card)
         mobo = mobo[0]
+    def is_a_product(component: dict):
```

Comment:
Please add newlines between code blocks to improve readability. 
```python
if xxx:
    pass

def func:
    ...
```

Also, rename `is_a_product` to a more standard `is_product`.